### PR TITLE
Test release version in autopilot tests

### PR DESCRIPTION
**Pull request by makhov**
Signed-off-by: Alexey Makhov <amakhov@mirantis.com>

## Description

Tests against several release versions of k0s are added. 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## How Has This Been Tested?

- [ ] Manual test
- [x] Auto test added

## Checklist:

- [x] My code follows the style [guidelines](https://github.com/k0sproject/k0s/blob/main/docs/contributors/overview.md) of this project 
- [ ] My commit messages are [signed-off](https://github.com/k0sproject/k0s/blob/main/docs/contributors/github_workflow.md)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -235,6 +235,13 @@ jobs:
         with:
           name: k0s
 
+      - name: Download latest release
+        if: matrix.smoke-suite == 'check-ap-ha3x3'
+        run: |
+          RELEASE_VERSION=`curl --silent "https://api.github.com/repos/k0sproject/k0s/releases/latest" | grep '"tag_name":' |  sed -E 's/.*"([^"]+)".*/\1/'`
```
Comment (s0j) [inline .github/workflows/go.yml:241]: Isn't the query against `/latest` sensitive to the ordering of releases in the releases page?  Does this need something like `k0s_sort`?

File: inttest/ap-ha3x3/ha3x3_test.go [hunk 2]
```diff
@@ -117,26 +119,17 @@ spec:
   timestamp: now
   commands:
     - k0supdate:
-        version: v0.0.0
-        forceupdate: true
+        version: ` + s.K0sUpdateVersion + `
         platforms:
           linux-amd64:
-            url: http://localhost/dist/k0s
+            url: http://localhost/dist/k0s-new
```
Comment (s0j) [inline inttest/ap-ha3x3/ha3x3_test.go:125]: There should probably be an assert in this test comparing the actual version of the new binary to the old version.

File: .github/workflows/go.yml [hunk 3]
```diff
@@ -272,6 +283,96 @@ jobs:
...
+          go-version: ${{ env.GO_VERSION }}
+
+      - id: set-matrix
+        run: |
+          set -x 
+          go install github.com/k0sproject/version/cmd/k0s_sort@v0.2.2
+          GOBIN="$(go env GOPATH)/bin"
+          matrix="["
+          COMMA=""
+          for i in 1.24.2 1.24.3; do \
```
Comment (s0j) [inline .github/workflows/go.yml:306]: The only problem in building up code in GHA yaml is that it becomes harder to catch things locally, before it gets to GHA. So to re-run a failing matrix test scenario would be something like:
* manually download the base k0s binary (`K0S_PATH`)
* have a compiled k0s (`K0S_UPDATE_FROM_PATH`)
* define the update version (`K0S_UPDATE_TO_VERSION`) 
* `make -C inttest check-ap-ha3x3`

Is there any way to move some of this in scripts to make it a bit easier for devs?

File: inttest/ap-ha3x3/ha3x3_test.go [hunk 4]
```diff
@@ -117,26 +117,17 @@ spec:
...
         targets:
           controllers:
             discovery:
-              static:
-                nodes:
-                  - controller0
-                  - controller1
-                  - controller2
+              selector: {}
```
Comment (s0j) [inline inttest/ap-ha3x3/ha3x3_test.go:127]: Does this need to be changed to selectors? The existing plan was intentionally extremely static, and now this becomes sensitive to the creation of the footloose cluster.

Comment (makhov) [reply inttest/ap-ha3x3/ha3x3_test.go:127]: Sorry, will change it back to static

File: .github/workflows/go.yml (hunk 1 above)
Comment (makhov) [reply .github/workflows/go.yml]: Fixed

File: inttest/ap-ha3x3/ha3x3_test.go (hunk 2 above)
Comment (makhov) [reply inttest/ap-ha3x3/ha3x3_test.go:123]: Done

File: .github/workflows/go.yml (hunk 3 above)
Comment (makhov) [reply .github/workflows/go.yml]: I've added `hack/tools/run-autopilot-matrix-tests.sh`, I hope it will help

**Review by s0j — APPROVED** [review]

**Pull request merged**